src/main/java/com/miraisolutions/xlconnect/Workbook.java

```diff
@@ -1490,18 +1535,19 @@ public int getLastColumn(String sheetName) {
         return getLastColumn(getSheet(sheetName));
     }
 
-    public void appendNamedRegion(DataFrame data, String name, boolean header) {
+    public void appendNamedRegion(DataFrame data, String name, String worksheetScope, boolean header) {
         Sheet sheet = workbook.getSheet(getName(name).getSheetName());
```

Comment:
Propagate `worksheetScope`
```suggestion
        Sheet sheet = workbook.getSheet(getNameInWorksheetScope(worksheetScope, name).getSheetName());
```